### ISSUE TITLE
Saved intelligence configurations

### ISSUE THREAD

**Issue by psugihara** [psugihara/FreeChat#43]
Dialing in your perfect settings is annoying and I don't want to have to do the same ones twice.

I want something like Terminal's theme picker (pictured below) but for different configurations of the Intelligence settings.

- Should save as Core Data model `Intelligence`
- Initialize with a few example configurations
- One Intelligence active at a time. In the future it could be cool to let the user easily swap on the new message screen.

<img width="779" alt="Screenshot 2023-12-21 at 10 25 42 AM" src="https://github.com/psugihara/FreeChat/assets/282016/2d65a1a9-b872-4708-9f1a-1ec10d78241e">
<img width="699" alt="Screenshot 2023-12-21 at 10 27 42 AM" src="https://github.com/psugihara/FreeChat/assets/282016/c7dd216c-ae73-449a-8819-043d61a2257e">

**Comment by EvolvingSoftware** [comment]
Is this the same as this idea from the readme? "Personas - save system prompt / model settings for later and change personas when you create a new conversation"

I was thinking something like the ability to perhaps create a folder structure for the Chats & use the folder to be able to craft a unique system prompt for each chat created in that folder (or first message automatically submitted for models that don't support a system prompt like: https://huggingface.co/bartowski/gemma-2-27b-it-GGUF)

**Comment by psugihara** [comment]
> Is this the same as this idea from the readme? "Personas - save system prompt / model settings for later and change personas when you create a new conversation"

Yeah this is an idea for how personas could be setup, saved, and selected.

Folder structure sounds like an interesting way to organize the chats, is that the idea? For my use-cases I think a simple tagged list is more immediate (though I really want search!). It looks like Notes has some optional folder functionality using NavigationSplitView that we could draw inspiration from.

<img width="1357" alt="Screenshot 2024-07-02 at 7 31 12 AM" src="https://github.com/psugihara/FreeChat/assets/282016/fd86d687-e196-4d67-8b5b-434d5701df0a">

Feels like there are a bunch of cool ways you could use/organize the different configurations and chats (and we should keep talking about them!) but probably the prerequisite is having something like the above to set them up.
